client/allocwatcher/alloc_watcher.go

```diff
@@ -279,15 +279,7 @@ func (p *localPrevAlloc) Migrate(ctx context.Context, dest *allocdir.AllocDir) e
 
 	p.logger.Debug("copying previous alloc")
 
-	moveErr := dest.Move(p.prevAllocDir, p.tasks)
-
-	// Always cleanup previous alloc
-	if err := p.prevAllocDir.Destroy(); err != nil {
```

Comment:
This removed a call to `p.prevAllocDir.UnmountAll()` as well, which among other benign effects, leaves `/secrets` laying around. I think we should add it back in.

Reply:
Doesn't that get cleaned up with the rest of the allocation GC? I guess I'm not sure why we special case removing bits of the previous allocdir, but only if we've migrated data. Ex. if we replaced the allocation without migration we wouldn't be doing anything to the previous allocdir that wasn't done as part of that previous allocation's lifecycle.

Reply:
Oh good point. Not sure why this was special cased.